File: lang/c/test/fun/vac/algorithms/sorting/QuickSortTest.c
#include <fun/vac/algorithms/sorting/QuickSort.h>
#include <fun/vac/util/SequenceBuilder.h>
#include <fun/vac/util/Sequences.h>
#include <fun/vac/util/TestRunner.h>

bool testQuickSort(void) {
    size_t size = 32768;

    int arr[size];
    SequenceBuilder_packRandom(arr, size);

    int checksum = Sequences_parityChecksum(arr, size);

    QuickSort_sort(arr, size);

    if (Sequences_parityChecksum(arr, size) != checksum) {
        return false;
    }

    if (!Sequences_isSorted(arr, size)) {
        return false;
    }

    return true;
}

int main(void) {
    TestRunner_parseTest(testQuickSort());
}

File: go/muse/algorithms/sorting/QuickSort/QuickSort_test.go
package QuickSort

import (
	"testing"

	"muse/util/SequenceBuilder"
	"muse/util/Sequences"
)

func TestQuickSort(t *testing.T) {
	size := 32768

	arr := make([]int, size)
	SequenceBuilder.PackRandom(arr)

	checksum := Sequences.ParityChecksum(arr)

	Sort(arr)

	if Sequences.ParityChecksum(arr) != checksum {
		t.FailNow()
	}

	if !Sequences.IsSorted(arr) {
		t.FailNow()
	}
}
